Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1560,4 +1561,46 @@ void Document::invalidate_stacking_context_tree()
         const_cast<Painting::PaintableBox*>(paint_box)->invalidate_stacking_context();
 }
 
+void Document::check_favicon_after_loading_link_resource()
+{
+    // https://html.spec.whatwg.org/multipage/links.html#rel-icon
+    //
+    // 4.6.7.8 Link type "icon"
+    //
+    // If multiple icons are provided, the user agent must select the most appropriate icon
+    // according to the type, media, and sizes attributes.
+    //
+    // If there are multiple equally appropriate icons, user agents must use the last one declared
+    // in tree order at the time that the user agent collected the list of icons.
```

Comment:
These comments can probably be put closer to where this logic is actually happening.
For example this part could be by the `favicon_link_elements->item(favicon_link_elements->length() - 1);` part

Reply:
Tried my best to improve comment placement, but I think some parts may be arguable.